File: runtime/src/main/java/ch/megard/akka/http/cors/javadsl/model/HttpHeaderRange.java
package ch.megard.akka.http.cors.javadsl.model;

import ch.megard.akka.http.cors.scaladsl.model.HttpHeaderRange$;
import org.apache.pekko.http.impl.util.Util;


/**
 * @see HttpHeaderRanges for convenience access to often used values.
 */
public abstract class HttpHeaderRange {
    public abstract boolean matches(String header);

    public static HttpHeaderRange create(String... headers) {
        return HttpHeaderRange$.MODULE$.apply(Util.convertArray(headers));
    }
}

File: runtime/src/main/scala/ch/megard/akka/http/cors/scaladsl/model/HttpHeaderRange.scala
package ch.megard.akka.http.cors.scaladsl.model

import ch.megard.akka.http.cors.javadsl

import java.util.Locale

abstract class HttpHeaderRange extends javadsl.model.HttpHeaderRange

object HttpHeaderRange {
  case object `*` extends HttpHeaderRange {
    def matches(header: String) = true
  }

  final case class Default(headers: Seq[String]) extends HttpHeaderRange {
    val lowercaseHeaders: Seq[String] = headers.map(_.toLowerCase(Locale.ROOT))
    def matches(header: String): Boolean = lowercaseHeaders contains header.toLowerCase(Locale.ROOT)
  }

  def apply(headers: String*): Default = Default(Seq(headers: _*))
}
